### PR TITLE
Cleaned up Tags

### PR DESCRIPTION
**Pull request by tom-pytel**
This is for merging after #143 (it does include all the changes from that PR which has already been approved but not yet merged). This cleans up the Tag mechanism as follows:
* Each Tag is now its own individual class, which allows 'overridable' to be property of the Tag and not specified at tag instantiate.
* Should be more performant.
* Key changes `url` -> `http.url` and `status.code` -> `http.status.code` to match Node agent.

### CODE REVIEW

**Comment by tom-pytel** [comment]
Do not merge until #144 is merged and I update this for it.

File: skywalking/trace/span.py
Comment (tom-pytel) [reply skywalking/trace/span.py:60]: ignore

**Comment by tom-pytel** [comment]
Or merge this first then update #144...

**Comment by tom-pytel** [comment]
Key @kezhenxu94, I am blocking on this a little, can I merge an move on?

**Comment by kezhenxu94** [comment]
> Key @kezhenxu94, I am blocking on this a little, can I merge an move on?

Oops, sorry forgot this. 🤦🏻

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**
